### ISSUE TITLE
Order date update is not updated when address is changed

### ISSUE THREAD

**Issue by hiousi** [PrestaShop/PrestaShop#13537]
**Describe the bug**

**To Reproduce**
1. Go to customer/address
2. Click on "edit" an address then save. 
3. check order.date_upd with sql for the related order.
4.  and see that date is not updated

**Screenshots**
If applicable, add screenshots or screenrecords to help explain your problem.

**Additionnal information**
PrestaShop version: all
PHP version: all

**Comment by marionf** [comment]
@hiousi

If you want to update the order.date_upd column, you need to change the address directly in the order and not in Customer > Address

**Comment by khouloudbelguith** [comment]
@hiousi, yes as said @marionf, the order.date_upd is updated only if we change address from the order page & not editing the address from BO or FO.
Thanks!

**Comment by hiousi** [comment]
@marionf @khouloudbelguith 
I'm happy to hear that we are all saying the same thing:
"_Prestashop provides a way of changing orders  without updating date_upd_"

**this is an unwanted behavior** (do not call it a bug if its too much for you)

- When an address is changed/modified for an order the order.date_upd should be updated.
or
- Prestashop should not allow an address to be modified when an order depends on it.
or
- customer/address page should be deleted or made not usable.

@marionf your proposal is a clever way of ignoring/reversing the issue. We have talked about the problem already there : https://github.com/PrestaShop/PrestaShop/pull/12963 and there https://github.com/PrestaShop/PrestaShop/pull/10847#event-2154468884.
Users can  change addresses without even knowing the existence of date_upd column and that an old order is linked to that address.

This issue is part of major other issues regarding how address are handled in Prestashop. (https://github.com/PrestaShop/PrestaShop/issues/13539).
